client/me/purchases/purchase-item/index.jsx

```diff
@@ -417,6 +418,15 @@ class PurchaseItem extends Component {
 			);
 		}
 
+		if ( ! isRechargeable( purchase ) && hasPaymentMethod( purchase ) ) {
```

Comment:
🤔 I think this condition _might_ be too permissive. I'm thinking of things like one-time purchases which are not renewable. I'm not 100% sure that this function would be called in that case, but if it is, then the message would be misleading.

If I'm right, we might just try adding `purchase.isAutoRenewEnabled &&` to the condition which then would be more similar to the above block.